lib/grant-types/abstract-grant-type.js

```diff
@@ -34,9 +35,12 @@ function AbstractGrantType(options) {
  * Generate access token.
  */
 
-AbstractGrantType.prototype.generateAccessToken = function() {
+AbstractGrantType.prototype.generateAccessToken = function(client, user, scope) {
   if (this.model.generateAccessToken) {
-    return Promise.try(this.model.generateAccessToken);
+    return Promise.try(promisify(this.model.generateAccessToken, 3), [client, user, scope])
+      .then(function(accessToken) {
+        return accessToken || tokenUtil.generateRandomToken();
```

Comment:
Do you think it's a good idea to fall back to the default `generateRandomToken`?
I would consider it an error if `model#generateAccessToken` is implemented but returns a falsy value.